lib/libc/sys/procctl.2

```diff
@@ -756,6 +756,42 @@ If the currently executed image in the process itself has revocation enabled,
 the
 .Dv PROC_CHERI_REVOKE_ACTIVE
 flag is or-ed with the value listed above.
+.It Dv PROC_CHERI_C18N_CTL
```

Comment:
Perhaps consider this commit into two commits: one that adds the new AT_BSDFLAGS flags value and updates rtld to check it, and a second commit that adds the procctl controls.  I think that split works nicely with your subsequent commits to add additional methods for setting the flag in AT_BSDFLAGS.